VultisigApp/VultisigApp/Services/Blowfish/BlowfishService.swift

```diff
@@ -44,6 +44,54 @@ struct BlowfishService {
         return response
     }
     
+    enum DecodingCustomError: Error {
+        case dataCorrupted(DecodingError.Context)
+        case keyNotFound(CodingKey, DecodingError.Context)
+        case valueNotFound(Any.Type, DecodingError.Context)
+        case typeMismatch(Any.Type, DecodingError.Context)
+    }
+
+    func scanSolanaTransactions
+    (
+        userAccount: String,
+        origin: String,
+        transactions: [String]
+    ) async throws -> BlowfishResponse? {
+        
+        let blowfishRequest = BlowfishSolanaRequest(
+            userAccount: userAccount,
+            metadata: BlowfishSolanaRequest.BlowfishMetadata(origin: origin),
+            transactions: transactions
+        )
+        
+        let endpoint = Endpoint.fetchBlowfishSolanaTransactions()
+        let headers = ["X-Api-Version" : "2023-06-05"]
+        let body = try JSONEncoder().encode(blowfishRequest)
+        let dataResponse = try await Utils.asyncPostRequest(urlString: endpoint, headers: headers, body: body)
+        
+        do {
+            let response = try JSONDecoder().decode(BlowfishResponse.self, from: dataResponse)
+            return response
+        } catch let DecodingError.dataCorrupted(context) {
+            print("Data corrupted: \(context)")
+            throw DecodingCustomError.dataCorrupted(context)
+        } catch let DecodingError.keyNotFound(key, context) {
```

Comment:
why you handling all decoding errors?

Reply:
It is because they don't have a typed payload, and I never know what is coming and or what is missing. So that helps whenever it breaks.